modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormDisplayContext.java

```diff
@@ -563,56 +563,47 @@ public boolean isAutosaveEnabled() throws PortalException {
 	}
 
 	public boolean isFormAvailable() throws PortalException {
-		if (isPreview()) {
-			return true;
-		}
+		DDMFormInstance ddmFormInstance = getFormInstance();
 
-		DDMFormInstance formInstance = getFormInstance();
+		if ((ddmFormInstance == null) || !isFormPublished()) {
+			return false;
+		}
 
-		if (formInstance != null) {
-			Group group = _groupLocalService.getGroup(
-				formInstance.getGroupId());
+		Group group = _groupLocalService.getGroup(ddmFormInstance.getGroupId());
 
-			Group scopeGroup = _groupLocalService.getGroup(
-				_portal.getScopeGroupId(_renderRequest));
+		Group scopeGroup = _groupLocalService.getGroup(
+			_portal.getScopeGroupId(_renderRequest));
 
-			if ((group != null) && (scopeGroup != null) &&
-				group.isStagingGroup() && !scopeGroup.isStagingGroup()) {
+		if ((group != null) && (scopeGroup != null) && group.isStagingGroup() &&
+			!scopeGroup.isStagingGroup()) {
 
-				return false;
-			}
+			return false;
+		}
 
-			if ((group != null) && group.isStagedRemotely()) {
-				ThemeDisplay themeDisplay = getThemeDisplay();
+		if ((group != null) && group.isStagedRemotely()) {
+			ThemeDisplay themeDisplay = getThemeDisplay();
 
-				Role role = _roleLocalService.getRole(
-					themeDisplay.getCompanyId(), RoleConstants.ADMINISTRATOR);
+			Role role = _roleLocalService.getRole(
+				themeDisplay.getCompanyId(), RoleConstants.ADMINISTRATOR);
 
-				List<User> users = _userLocalService.getRoleUsers(
-					role.getRoleId());
+			List<User> users = _userLocalService.getRoleUsers(role.getRoleId());
 
-				if (!DDMFormInstanceStagingUtil.
-						isFormInstancePublishedToRemoteLive(
-							group, users.get(0), formInstance.getUuid())) {
+			if (!DDMFormInstanceStagingUtil.isFormInstancePublishedToRemoteLive(
+					group, users.get(0), ddmFormInstance.getUuid())) {
 
-					return false;
-				}
+				return false;
 			}
 		}
 
 		if (isSharedURL()) {
-			if (isFormPublished() && isFormShared()) {
+			if (isFormShared()) {
 				return true;
 			}
 
 			return false;
 		}
 
-		if (formInstance != null) {
-			return true;
-		}
-
-		return false;
+		return true;
```

Comment:
@leticia-maciel can you please check if we can change to:

```
if (!isFormShared() && isSharedURL()) {
     return false;
}

return true;
```
Thanks!

Reply:
I'll test it now

Reply:
Done!